manager/director/apps/sites/forms.py

```diff
@@ -161,6 +171,35 @@ def clean(self) -> Dict[str, Any]:
             if "domain" in cleaned_data and cleaned_data["domain"].endswith("tjhsst.edu"):
                 self.add_error("domain", "Only administrators can add tjhsst.edu domains")
 
+        # Check if this domain CNAMEs to the proper value
+        if "domain" in cleaned_data and cleaned_data["domain"] and not self.user_is_superuser:
```

Comment:
```suggestion
        if cleaned_data.get("domain") and not self.user_is_superuser:
```

This is what I would probably do. I think it's a little more readable because it helps cut down the line length, but that's just my opinion. Feel free to ignore this.